persistence/src/test/java/com/epam/rft/atsy/persistence/repositories/PasswordHistoryRepositoryIT.java

```diff
@@ -0,0 +1,139 @@
+
+    @Autowired
+    private PasswordHistoryRepository repository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Test
+    public void findByUserEntityShouldNotFindPasswordHistoryForUserWithoutPasswordHistory() {
+        //Given
+        UserEntity user = this.userRepository.findByUserName(TEST_USERNAME_WITHOUT_PASSWORD_HISTORY);
+
+        //When
+        List<PasswordHistoryEntity> result = this.repository.findByUserEntity(user);
+
+        //Then
+        assertThat(result, notNullValue());
+        assertThat(result, empty());
+    }
+
+    @Test
+    public void findByUserEntityShouldFindSinglePasswordHistoryForUserWithSinglePasswordHistory() {
+        //Given
+        UserEntity user = this.userRepository.findByUserName(TEST_USERNAME_WITH_SINGLE_PASSWORD_HISTORY);
+        Date nearNow = currentDateMinus(5);
+
+        //When
+        List<PasswordHistoryEntity> result = this.repository.findByUserEntity(user);
+
+        //Then
+        assertThat(result, notNullValue());
+        assertThat(result.size(), is(1));
+        assertPasswordHistoryEntity(result.get(0), user, user.getUserPassword(), nearNow);
+    }
+
+    @Test
+    public void findByUserEntityShouldFindThreePasswordHistoryForUserWithThreePasswordHistory() {
+        //Given
+        UserEntity user = this.userRepository.findByUserName(TEST_USERNAME_WITH_THREE_PASSWORD_HISTORY);
+        Date nearNow = currentDateMinus(5);
+
+        //When
+        List<PasswordHistoryEntity> result = this.repository.findByUserEntity(user);
+
+        //Then
+        assertThat(result, notNullValue());
+        assertThat(result.size(), is(3));
+
+        assertPasswordHistoryEntity(result.get(0),
+                user,
+                user.getUserPassword(),
+                nearNow);
+
+        assertPasswordHistoryEntity(result.get(1),
+                user,
+                TEST_PASSWORD_1_FOR_USER_WITH_THREE_PASSWORD_HISTORY,
+                nearNow);
+
+        assertPasswordHistoryEntity(result.get(2),
+                user,
+                TEST_PASSWORD_2_FOR_USER_WITH_THREE_PASSWORD_HISTORY,
+                nearNow);
+
+    }
+
+    @Test
+    public void findOldestPasswordShouldNotFindPasswordHistoryForUserWithoutPasswordHistory() {
+        //When
+        PasswordHistoryEntity pwHistory = this.repository.findOldestPassword(TEST_USER_ID_WITHOUT_PASSWORD_HISTORY);
+
+        //Then
+        assertThat(pwHistory, nullValue());
+    }
+
+    @Test
+    public void findOldestPasswordShouldFindOldestPasswordHistoryForUserWithThreePasswordHistory() {
+        //Given
+        UserEntity user = this.userRepository.findByUserName(TEST_USERNAME_WITH_THREE_PASSWORD_HISTORY);
+        Date nearNow = currentDateMinus(5);
+
+        //When
+        PasswordHistoryEntity pwHistory = this.repository.findOldestPassword(TEST_USER_ID_WITH_THREE_PASSWORD_HISTORY);
```

Comment:
As there can be other test who insert users as test data, the hard coded user id values can go wrong in the future.

I suggest to remove all TEST_USER_ID_... constants from the test and use id from the user returned by userRepository.findByUserName().